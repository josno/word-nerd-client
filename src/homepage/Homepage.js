import React, { Component } from 'react';
import EntryOption from '../entryoption/EntryOption';
import './Homepage.css';

class Homepage extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}
	render() {
		return (
			<div>
				<header role="banner">
					<h1 className="home-page-title"> Word Nerd </h1>
				</header>
				<section>
					<header>
						<h3>
							Make english games to play in the classroom
							together.
						</h3>
					</header>
					<p>
						Simply type in vocabulary words you'd like to practice
						for fluency and generate a "Pass the Ball" game to play
						together in the classroom.
					</p>
				</section>
				<section>
					<EntryOption />
				</section>
			</div>
		);
	}
}

export default Homepage;
