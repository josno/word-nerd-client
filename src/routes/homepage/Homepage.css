.app-description-container {
	margin: 0 2em 0 2em;
}

.homepage {
	background: #18c0f0;
	height: 100vh;
}
.home-page-title {
	color: #ffbc42;
	font-family: 'Lemon', sans-serif;
	font-size: 3rem;
	margin: 0;
	padding-top: 0.5em;
	text-shadow: 3px 3px #ff206e;
	-webkit-text-stroke: 1px black;
}

@media (min-width: 400px) {
	body {
		font-size: 18px;
	}

	.home-page-title {
		font-size: 4rem;
	}
}

@media (min-width: 700px) {
	body {
		font-size: 22px;
	}

	.home-page-title {
		font-size: 6rem;
	}
}
