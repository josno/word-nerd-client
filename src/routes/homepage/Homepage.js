import React, { Component } from 'react';
import EntryOption from '../../components/EntryOption/EntryOption';
import './Homepage.css';

class Homepage extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}
	render() {
		return (
			<div>
				<header role="banner">
					<h1 className="home-page-title"> Word Nerd </h1>
				</header>
				<section>
					<header>
						<h3>
							Make english games to play in the classroom
							together.
						</h3>
					</header>
					<p>
						Generate a "Pass the Ball" game to play together in the
						classroom using your own vocabulary words.
					</p>
				</section>
				<section>
					<EntryOption />
				</section>
			</div>
		);
	}
}

export default Homepage;
