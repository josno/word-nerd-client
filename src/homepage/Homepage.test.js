import React from 'react';
// import ReactDOM from 'react-dom';
// import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Homepage from './Homepage';

describe('<Home/>', () => {
	it('renders as expected', () => {
		const tree = renderer.create(<Homepage />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
