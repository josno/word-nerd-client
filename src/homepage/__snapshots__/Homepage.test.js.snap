// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> renders as expected 1`] = `
<div>
  <header
    role="banner"
  >
    <h1
      className="home-page-title"
    >
       Word Nerd 
    </h1>
  </header>
  <section>
    <header>
      <h3>
        Make english games to play in the classroom together.
      </h3>
    </header>
    <p>
      Generate a "Pass the Ball" game to play together in the classroom using your own vocabulary words.
    </p>
  </section>
  <section>
    <div
      className="entry-option"
    >
      <div
        className="form"
      >
        <ul>
          <li
            className="form-tab-list-item"
          >
            <button
              className="form-tab"
              onClick={[Function]}
            >
              Sign Up
            </button>
          </li>
          <li
            className="form-tab-list-item"
          >
            <button
              className="form-tab"
              onClick={[Function]}
            >
              Login
            </button>
          </li>
        </ul>
        <div
          className="sign-up"
        >
          <section>
            <div
              className="input-form"
            >
              <input
                className="form-input"
                htmlFor="first-name"
                name="firstName"
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />
              <input
                className="form-input"
                htmlFor="last-name"
                name="lastName"
                onChange={[Function]}
                placeholder="Last Name"
                type="text"
                value=""
              />
              <input
                className="form-input"
                htmlFor="username"
                name="username"
                onChange={[Function]}
                placeholder="username"
                type="text"
                value=""
              />
              <input
                className="form-input"
                htmlFor="password"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="text"
                value=""
              />
            </div>
            <a
              href="/game-home-page"
              onClick={[Function]}
            >
              <button
                className="input-submit-button"
              >
                Sign Up
              </button>
            </a>
          </section>
        </div>
      </div>
    </div>
  </section>
</div>
`;
